includes/class-wc-skroutz-analytics-product.php

```diff
@@ -77,6 +77,39 @@ public function get_id() {
 		return apply_filters( 'wc_skroutz_analytics_product_id_filter', $product_id, $this->product );
 	}
 
+	/**
+	 * Returns the product variation id that should be reported to Analytics based on
+	 * product id admin settings.
+	 *
+	 * @return string|integer  The product variation id that should be reported to Analytics
+	 *
+	 * @since    1.7.0
+	 */
+	public function get_variable_id() {
+		$product_id = $this->get_custom_product_id($this->product);
+
+		if ( $this->product->is_type( 'variable' ) && $this->items_product_id_settings['parent_id_enabled'] == 'no' ) {
+			$this->product = wc_get_product($this->product->get_children()[0]);
```

Comment:
A product can be non visible (enabled property is not checked on admin panel). In order to increase our chances we should get the first visible product id:

```
// TODO Use only get_visible_children when we drop support for WooCommerce < 3.0
if ( method_exists( $this->product, 'get_visible_children' ) ) {
  $this->product = wc_get_product($this->product->get_visible_children()[0]);
} else {
  $this->product = wc_get_product($this->product->get_children(true)[0]);
}
```